src/time-slot-bar/time-slot-bar.tsx

```diff
@@ -0,0 +1,307 @@
+import { useEffect, useRef, useState } from "react";
+import {
+    ArrowButton,
+    ArrowIconLeft,
+    ArrowIconRight,
+    Border,
+    CELL_WIDTH,
+    CellText,
+    Container,
+    TimeLabel,
+    TimeMarker,
+    TimeMarkerWrapper,
+    TimeSlot,
+    TimeSlotBarContainer,
+    TimeSlotWrapper,
+} from "./time-slot-bar.styles";
+import { TimeSlotBarProps } from "./types";
+// Load plugins
+dayjs.extend(customParseFormat);
+
+export const TimeSlotBar = ({
+    testId,
+    startTime,
+    endTime,
+    slots,
+    onClickSlot,
+    defaultTimeSlot,
+}: TimeSlotBarProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const barRef = useRef<HTMLDivElement>(null);
+    const [scrollPosition, setScrollPosition] = useState<number>(0);
+    const [clientWidth, setClientWidth] = useState<number>(0);
+
+    const CELL_DURATION = 30; // In minutes
+    const SCROLL_INCREMENT = CELL_WIDTH * 2.5; // In px. Each scroll increment corresponds to 75mins
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+
+    useEffect(() => {
+        const handleScroll = () => {
+            if (barRef.current) {
+                setScrollPosition(barRef.current.scrollLeft);
+            }
+        };
+
+        const container = barRef.current;
+        if (container) {
+            window.addEventListener("resize", handleResize);
+            container.addEventListener("scroll", handleScroll);
+            // Load initial values
+            handleResize();
+        }
+
+        return () => {
+            if (container) {
+                window.removeEventListener("resize", handleResize);
+                container.removeEventListener("scroll", handleScroll);
+            }
+        };
+    }, []);
+
+    // =============================================================================
+    // HANDLERS
+    // =============================================================================
+
+    const handleScroll = (direction: "left" | "right") => {
+        if (barRef.current) {
+            barRef.current.scrollBy({
+                left:
+                    direction === "left" ? -SCROLL_INCREMENT : SCROLL_INCREMENT,
+                behavior: "smooth",
+            });
+        }
+    };
+
+    const handleResize = () => {
+        if (barRef.current) {
+            setClientWidth(barRef.current.clientWidth);
+        }
+    };
+
+    // ===========================================================================
+    // HELPER FUNCTIONS
+    // ===========================================================================
+
+    // Helper function to calculate the difference in minutes between two times
+    const getTimeDiffInMinutes = (start: string, end: string) => {
```

Comment:
this function can be moved outside of the component to prevent recreation during a re-render. Possibly to `DateHelper` for shared usage